### ISSUE TITLE
Dump/Restore does not work

### ISSUE THREAD

**Issue by uvchik** [oemof/oemof#186]
The dump and restore method of the EnergySystem class does not work in the release branch of v0.1

There is no error message but if you uncomment dump/restore at the button of the storage_invest.py example there will be an error processing the results. This means that the structure of the results after the restore is different than before.

``` python
Traceback (most recent call last):
  File "/home/uwe/rli-lokal/git_home/oemof/examples/solph/storage_optimization/storage_invest.py", line 234, in <module>
    pp.pprint(get_result_dict(esys))
  File "/home/uwe/rli-lokal/git_home/oemof/examples/solph/storage_optimization/storage_invest.py", line 142, in get_result_dict
    myresults = outputlib.DataFramePlot(energy_system=energysystem)
  File "/home/uwe/rli-lokal/git_home/oemof/oemof/outputlib/__init__.py", line 207, in __init__
    super(DataFramePlot, self).__init__(**kwargs)
  File "/home/uwe/rli-lokal/git_home/oemof/oemof/outputlib/__init__.py", line 88, in __init__
12:55:41-INFO-Check the results
    row['bus_label'] = list(k.outputs.keys())[0].label
IndexError: list index out of range
```

**Comment by ckaldemeyer** [comment]
This might be an issue with pickle!?

**Comment by uvchik** [comment]
Fixed by #231.
